src/analyze.ts

```diff
@@ -123,10 +123,42 @@ export function findExports (code: string): ESMExport[] {
       exp.names = [exp.name]
     }
   }
-
+  const exportsLocation = getExportTokenLocation(code)
   return exports.filter((exp, index, exports) => {
+    // Filter out noise that does not match the semantics of export
+    if (!isExportStatement(exportsLocation, exp)) {
+      return false
+    }
     // Prevent multiple exports of same function, only keep latest iteration of signatures
     const nextExport = exports[index + 1]
-    return !nextExport || exp.type !== nextExport.type || !exp.name || exp.name !== nextExport.name
+    return isExportStatement(exportsLocation, exp) || !nextExport || exp.type !== nextExport.type || !exp.name || exp.name !== nextExport.name
```

Comment:
Perf: Put more complex checks to the end:

```suggestion
    return !nextExport || exp.type !== nextExport.type || !exp.name || exp.name !== nextExport.name || isExportStatement(exportsLocation, exp)
```

Reply:
Putting it at the end should not be feasible, because it is an or judgment, if there is a correct option before it, the later judgment will not be executed and some cases will be missed

Reply:
Logically `a || b || c` should be same as `c || b || a` i guess :) But when `a` is an statement, runtime early returns.

Reply:
Sorry，If you'll excuse me, there should be no need to call `isExportStatement` in the or statement to determine if it's a legitimate export statement, as it's already been called before.